UACloudLibraryServer/Controllers/InfoModelController.cs

```diff
@@ -215,31 +212,62 @@ public IActionResult GetAllNamesandAddressSpacesAsync()
             return new ObjectResult("Upload successful!") { StatusCode = (int)HttpStatusCode.OK };
         }
 
+        private string RetrieveVersionFromNodeset(AddressSpace uaAddressSpace)
+        {
+            try
+            {
+                // workaround for bug https://github.com/dotnet/runtime/issues/67622
+                string nodesetXml = uaAddressSpace.Nodeset.NodesetXml.Replace("<Value/>", "<Value xsi:nil='true' />");
+
+                using (Stream stream = new MemoryStream(Encoding.UTF8.GetBytes(nodesetXml)))
+                {
+                    UANodeSet nodeSet = UANodeSet.Read(stream);
+                    if ((nodeSet.Models != null) && (nodeSet.Models.Length > 0))
+                    {
+                        // take the data from the first model
+                        ModelTableEntry model = nodeSet.Models[0];
+                        if (model != null)
+                        {
+                            return model.Version;
+                        }
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                return string.Empty;
+            }
+
+            return string.Empty;
+        }
+
         private void RetrieveDatesFromNodeset(AddressSpace uaAddressSpace, out DateTime publicationDate, out DateTime lastModifiedDate)
         {
             publicationDate = DateTime.UtcNow;
             lastModifiedDate = DateTime.UtcNow;
 
             try
             {
-                using (Stream stream = new MemoryStream(Encoding.UTF8.GetBytes(uaAddressSpace.Nodeset.NodesetXml)))
+                // workaround for bug https://github.com/dotnet/runtime/issues/67622
+                string nodesetXml = uaAddressSpace.Nodeset.NodesetXml.Replace("<Value/>", "<Value xsi:nil='true' />");
+
+                using (Stream stream = new MemoryStream(Encoding.UTF8.GetBytes(nodesetXml)))
                 {
                     UANodeSet nodeSet = UANodeSet.Read(stream);
                     if ((nodeSet.Models != null) && (nodeSet.Models.Length > 0))
                     {
-                        foreach (ModelTableEntry model in nodeSet.Models)
+                        // take the data from the first model

```

Comment:
Should this also take the info from the first model, like version?
Has there been discussion about supporting multiple models? I haven't run into any multi-model nodesets. Can we just disallow on upload?

Reply:
Good catch, yes, we should!

Reply:
Actually, we shouldn't update the hashing function to remain backward compatibility and it doesn't matter if we create the hash from all models in the nodeset file.

Reply:
It looks like the hashing function does its own enumeration over all models and shouldn't be impact by this?

Reply:
Correct. If you don't have any other concerns, please approve.